extensions/pdh/PerformanceDataMonitor.cpp

```diff
@@ -264,16 +281,29 @@ void PerformanceDataMonitor::setupMembersFromProperties(const std::shared_ptr<co
 
   std::string output_format_string;
   if (context->getProperty(OutputFormatProperty.getName(), output_format_string)) {
-    if (output_format_string == OPEN_TELEMETRY_FORMAT_STR) {
-      logger_->log_trace("OutputFormat is configured to be OpenTelemetry");
+    if (output_format_string == PRETTY_OPEN_TELEMETRY_FORMAT_STR || output_format_string == COMPACT_OPEN_TELEMETRY_FORMAT_STR) {
       output_format_ = OutputFormat::OPENTELEMETRY;
-    } else if (output_format_string == JSON_FORMAT_STR) {
-      logger_->log_trace("OutputFormat is configured to be JSON");
+      pretty_output_ = output_format_string == PRETTY_OPEN_TELEMETRY_FORMAT_STR;
+      logger_->log_trace("OutputFormat is configured to be %s OpenTelemetry", pretty_output_ ? "pretty" : "compact");
+    } else if (output_format_string == PRETTY_JSON_FORMAT_STR || output_format_string == COMPACT_JSON_FORMAT_STR) {
       output_format_ = OutputFormat::JSON;
+      pretty_output_ = output_format_string == PRETTY_JSON_FORMAT_STR;
+      logger_->log_trace("OutputFormat is configured to be %s JSON", pretty_output_ ? "pretty" : "compact");
     } else {
-      logger_->log_error("Invalid OutputFormat, defaulting to JSON");
       output_format_ = OutputFormat::JSON;
+      pretty_output_ = true;
+      logger_->log_error("Invalid OutputFormat, defaulting to %s JSON", pretty_output_ ? "pretty" : "compact");
+    }
+  }
+
+  std::string double_precision_string;
+  if (context->getProperty(DoublePrecisionProperty.getName(), double_precision_string)) {
```

Comment:
The getProperty should do the conversion implicitly, so reading this to `double_precision_` variable should be enough.

Reply:
Good idea, unfortunetly int8_t is not yet supported by Value.h and ValueParser.h (which the getProperty uses) but left TODO and will replace this in a separate PR if thats okay with you
(I didnt want to increase the scope of this PR with the int8_t Value implementation)

Reply:
Sure it's okay in a separate PR, thanks!